subprojects/com.mbeddr/build.gradle

```diff
@@ -153,25 +160,59 @@ private static void configureRepositories(Project project) {
 }
 
 
-task printVersions {
+tasks.register('printVersions') {
     doLast {
         println "mbeddrBuildNumber: $project.mbeddrBuildNumber"
         println "mbeddrPlatformBuildNumber: $project.mbeddrPlatformBuildNumber"
     }
 }
 
-task printRepositories {
+tasks.register('printRepositories') {
     doLast {
         println "snapshotRepository: $snapshotRepository"
         println "releaseRepository: $releaseRepository"
         println "dependencyRepositories: $dependencyRepositories"
     }
 }
 
-tasks.register('migrate') {
-    // No-op in this version
+List<String> projectsInDependencyOrder = [
+        'com.mbeddr.mpsutil',
+//        'com.mbeddr.doc',
+//        'com.mbeddr.doc.aspect',
+//        'com.mbeddr.core',
+//        'com.mbeddr.ext',
+//        'com.mbeddr.cc',
+//        'com.mbeddr.xmodel',
+        'com.mbeddr.build',
+]
+
+List<File> projectDirectoriesInDependencyOrder = projectsInDependencyOrder.collect {
+    new File(rootProject.file('code/languages'), it)
+}
+
+tasks.register('migrate', MpsMigrate) {
+    dependsOn(':com.mbeddr:platform:generate_platform_languages')
+    mpsHome = mpsHomeDir
+    projectDirectories.from(projectDirectoriesInDependencyOrder)
+
+    folderMacros['mbeddr.github.core.home'] = rootProject.layout.projectDirectory
+
+    pluginRoots.from(new File(mpsHomeDir, 'plugins'))
+
+    maxHeapSize = "4G"
 }
 
-tasks.register('remigrate') {
-    // No-op in this version
+tasks.register('remigrate', Remigrate) {
+    dependsOn(':com.mbeddr:platform:generate_platform_languages')
+    mpsHome = mpsHomeDir
+    projectDirectories.from(projectDirectoriesInDependencyOrder)
+
+    folderMacros['mbeddr.github.core.home'] = rootProject.layout.projectDirectory
+
+    pluginRoots.from(new File(mpsHomeDir, 'plugins'))
+
+    maxHeapSize = "4G"
```

Comment:
Is 4 GB really enough for mbeddr? I am probably I bit to cautious but shouldn't we set it to 8G to be sure?

Reply:
Yes, if it wasn't enough the build would have failed.